source/image/ImageSet.cpp

```diff
@@ -57,61 +52,59 @@ namespace {
 
 	// Determine whether the given path or name is to a sprite for which a
 	// collision mask ought to be generated.
-	bool IsMasked(const string &path)
+	bool IsMasked(const filesystem::path &path)
 	{
-		if(path.length() >= 5 && path.compare(0, 5, "ship/") == 0)
-			return true;
-		if(path.length() >= 9 && path.compare(0, 9, "asteroid/") == 0)
-			return true;
-
-		return false;
+		filesystem::path directory = *path.begin();
```

Comment:
```suggestion
		if(path.empty())
			return false;
		filesystem::path directory = *path.begin();
```
just to be sure you're not dereferencing end()